internal/controller/ovnk_ipfix_collector.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2025 Illumio, Inc. All Rights Reserved.
+
+package controller
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"net"
+	"os"
+	"time"
+
+	pb "github.com/illumio/cloud-operator/api/illumio/cloud/k8sclustersync/v1"
+	netflows "github.com/netsampler/goflow2/decoders/netflow"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OVNKFlow represents a flow captured from OVN-Kubernetes.
+type OVNKFlow struct {
+	SourceIP        string
+	DestinationIP   string
+	SourcePort      uint16
+	DestinationPort uint16
+	Protocol        string
+	IPVersion       string
+	StartTimestamp  *timestamppb.Timestamp
+}
+
+// https://datatracker.ietf.org/doc/rfc7011/
+// The IPFIX Message Header 16-bit Length field limits the length of an
+// IPFIX Message to 65535 octets, including the header.  A Collecting
+// Process MUST be able to handle IPFIX Message lengths of up to
+// 65535 octets.
+const ipfixMaxLength = 65535
+
+// isOVNKDeployed checks for the presence of the `openshift-ovn-kubernetes` namespace, detection based on OVN-K namespace, this is configurable.
+// https://ovn-kubernetes.io/installation/launching-ovn-kubernetes-on-kind/#run-the-kind-deployment-with-podman
+func (sm *streamManager) isOVNKDeployed(logger *zap.Logger, ovnkNamespace string) bool {
+	logger.Debug("Checking for OVN-K deployment")
+	clientset, err := NewClientSet()
+	if err != nil {
+		logger.Error("Failed to create clientset", zap.Error(err))
+		return false
+	}
+	_, err = clientset.CoreV1().Namespaces().Get(context.TODO(), ovnkNamespace, metav1.GetOptions{})
```

Comment:
Pass a proper Context.
```suggestion
	_, err = clientset.CoreV1().Namespaces().Get(ctx, ovnkNamespace, metav1.GetOptions{})
```